src/main/java/com/redhat/labs/lodestar/service/GitService.java

```diff
@@ -0,0 +1,337 @@
+	 * 
+	 * @param project
+	 * @return
+	 */
+	ProjectTree createProjectTree(Project project) {
+
+		LOGGER.debug("creating project tree for {}", project);
+
+		List<ProjectTreeNode> treeNodes = getProjectTree(project.getId(), false);
+		return ProjectTree.builder().projectId(project.getId()).projectTreeNodes(treeNodes).build();
+
+	}
+
+	/**
+	 * Returns an {@link Optional} containing the {@link ProjectTreeNode} for the
+	 * artifacts.json file. Otherwise, an empty {@link Optional} is returned.
+	 * 
+	 * @param projectTree
+	 * @return
+	 */
+	Optional<File> getArtifactsFile(ProjectTree projectTree) {
+
+		Optional<ProjectTreeNode> artifactsTreeNode = projectTree.getArtifactsProjectTreeNode();
+		if (artifactsTreeNode.isEmpty()) {
+			return Optional.empty();
+		}
+
+		// get default branch
+		String branch = findDefaultBranch(projectTree.getProjectId());
+
+		return Optional.ofNullable(
+				gitlabRestClient.getFile(projectTree.getProjectId(), artifactsTreeNode.get().getPath(), branch));
+
+	}
+
+	/*
+	 * Returns a {@link List} of {@link Artifact}s for the given {@link File}. An
+	 * empty {@link List} is returned if the {@link File} is not found or contains
+	 * no {@link Artifact}s.
+	 */
+	List<Artifact> parseFile(Optional<File> file) {
+
+		List<Artifact> artifacts = new ArrayList<>();
+
+		if (file.isEmpty()) {
+			return artifacts;
+		}
+
+		File artifactsFile = file.get();
+
+		artifactsFile.decodeFileAttributes();
+
+		if (ARTIFACT_FILE.equals(artifactsFile.getFilePath())) {
+			artifacts = parseArtifactsFile(artifactsFile);
+		} else if (ENGAGEMENT_FILE.equals(artifactsFile.getFilePath())) {
+			artifacts = parseEngagementFile(artifactsFile);
+		}
+
+		return artifacts;
+
+	}
+
+	/**
+	 * Returns a {@link List} of {@link Artifact}s from the given {@link File}. An
+	 * empty {@link List} is returned if no file found or contains no
+	 * {@link Artifact}s.
+	 * 
+	 * @param file
+	 * @return
+	 */
+	List<Artifact> parseArtifactsFile(File file) {
+
+		if (null == file || null == file.getContent() || file.getContent().isBlank()) {
+			return new ArrayList<>();
+		}
+
+		return Arrays.asList(jsonb.fromJson(file.getContent(), Artifact[].class));
+
+	}
+
+	/**
+	 * Returns a {@link List} of {@link Artifact}s from the given {@link File}. An
+	 * empty {@link List} is returned if no file found or contains no
+	 * {@link Artifact}s.
+	 * 
+	 * @param file
+	 * @return
+	 */
+	List<Artifact> parseEngagementFile(File file) {
+
+		if (null == file || null == file.getContent() || file.getContent().isBlank()) {
+			return new ArrayList<>();
+		}
+
+		Engagement engagement = jsonb.fromJson(file.getContent(), Engagement.class);
+
+		// set engagement uuid on each artifacts
+		List<Artifact> artifacts = null == engagement.getArtifacts() ? new ArrayList<>() : engagement.getArtifacts();
+		artifacts.stream().forEach(a -> a.setEngagementUuid(engagement.getUuid()));
+
+		return artifacts;
+
+	}
```

Comment:
I don't think we should support engagement.json artifacts in this repo. When we cutover, a migration run should be done (thinking in git-api and each migration anded to a migration service we can rip out when completed. - with some on / off support or api invocation(?)

Reply:
no longer handling engagement.json